internal/resources/fetcher.go

```diff
@@ -58,12 +64,46 @@ func NewFetcher(kubewardenNamespace string, policyServerURL string) (*Fetcher, e
 // policies that should evaluate these resources. Example:
 // AuditableResources{Policies:[policy1, policy2] Resources:[podA, podB], Policies:[policy1] Resources:[deploymentA], Policies:[policy3] Resources:[ingressA]}
 func (f *Fetcher) GetResourcesForPolicies(ctx context.Context, policies []policiesv1.Policy, namespace string) ([]AuditableResources, error) {
+	return f.getResources(ctx, policies, namespace)
+}
+
+// Method to check if the given resource is namespaced or not.
+func (f *Fetcher) isNamespacedResource(gvr schema.GroupVersionResource) (bool, error) {
+	discoveryClient := f.clientset.Discovery()
+
+	apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(gvr.GroupVersion().String())
+	if err != nil {
+		return false, err
+	}
+	for _, apiResource := range apiResourceList.APIResources {
+		if apiResource.Name == gvr.Resource {
+			return apiResource.Namespaced, nil
+		}
+	}
+	return false, auditscannererrors.ErrResourceNotFound
+}
+
+// Method used by other method to fetch resources evaluated by the given policies.
+func (f *Fetcher) getResources(ctx context.Context, policies []policiesv1.Policy, namespace string) ([]AuditableResources, error) {
 	auditableResources := []AuditableResources{}
 	gvrMap := createGVRPolicyMap(policies)
 	for gvr, policies := range gvrMap {
+		if namespace == constants.NoNamespace {
+			isNamespaced, err := f.isNamespacedResource(gvr)
+			if err != nil {
+				if errors.Is(err, auditscannererrors.ErrResourceNotFound) {
+					log.Warn().Msg(fmt.Sprintf("API resource (%s) not found", gvr.String()))
+					continue
+				}
+				return nil, err
+			}
+			if isNamespaced {
+				continue
+			}
+		}
```

Comment:
I think this deserves to be placed into a dedicated method, like `getClusterwideResources`, with an explanation about why this is different from `getResources`